vnext/Desktop/WinHTTPWebSocketResource.cpp

```diff
@@ -0,0 +1,447 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+#include "pch.h"
+
+#include <Unicode.h>
+#include <Utils.h>
+#include "WinHTTPWebSocketResource.h"
+
+// Windows API
+#include <comutil.h>
+
+using Microsoft::Common::Unicode::Utf8ToUtf16;
+
+using std::function;
+using std::size_t;
+using std::string;
+using std::wstring;
+
+namespace Microsoft::React
+{
+
+WinHTTPWebSocketResource::WinHTTPWebSocketResource(const string& urlString)
+  : m_openSessionFlags{ WINHTTP_FLAG_ASYNC }
+  , m_sessionHandle{ NULL }
+  , m_connectionHandle{ NULL }
+  , m_requestHandle{ NULL }
+  , m_errorStatus{ ERROR_SUCCESS }
+{
+#if 0 // WinHttpCrackUrl does not accept non-HTTP schemes 🙄
+  BSTR bstr = _com_util::ConvertStringToBSTR(urlString.c_str());
+  LPWSTR lpwstr = bstr;
+  SysFreeString(bstr);
+
+  LPURL_COMPONENTS url{};
+  auto parseResult = WinHttpCrackUrl(lpwstr, static_cast<DWORD>(urlString.length()), /*flags*/ 0, url);
```

Comment:
this is the wrong usage of the API. The way you would use this api is as follows:
URL_COMPONENTS url { sizeof(URL_COMPONENTS) };
WinHttpCrackUrl(..., ..., &url);

Reply:
Noted.

For reference, the current usage still works, but the usage of `WinCrackUrl` has been discarded because it won't accept non-HTTP schemes (such as `ws://` or `wss://` 🙄)